### ISSUE TITLE
Installation fails in Manjaro

### ISSUE THREAD

**Issue by volpatto** [firedrakeproject/firedrake#1543]
Hi all,

I have been struggling with Firedrake installation with fresh install. Everything was fine, then I tried to update Firedrake and now I'm stuck due to PNETCDF error, which looks like a known issue (please see #1526).

I have tried some workarounds. Let me state everything.

### First attempt: a regular fresh install

I downloaded the install script and executed it as recommended in Firedrake's website:
```console
$ curl -O https://raw.githubusercontent.com/firedrakeproject/firedrake/master/scripts/firedrake-install
$ python3 firedrake-install --disable-ssh --no-package-manager
```
Since I'm using an Archlinux, I have to pass `--no-package-manager`. This is OK, I never had problems with it until now.

The result error is related to PNETCDF build, and can be checked in the log provided below.

[firedrake-install_fresh.log](https://github.com/firedrakeproject/firedrake/files/3849827/firedrake-install_fresh.log)

### Second attempt: use `mpich` instead of `openmpi`

AFAIK (based on #1526, although for a different OS), this problem is related to `openmpi`, and a proper replacement by `mpich` should work (at least it was my hope). Spoiler: failed.

This time, I tried to use `mpiexec.hydra`, passing it through install scripts args. My command was:

```console
$ python3 firedrake-install --disable-ssh --no-package-manager --mpiexec mpiexec.hydra --mpicc mpicc --mpicxx mpicxx --mpif90 mpif90
```

Now, I will give some remarks (that I don't understand entirely). First, I can't remove `openmpi`, since arch base depends on it, so it must have precedence over `mpich` (break my system is not worthy). Thus, in my PATH, `openmpi` must always be found before `mpich`. Some discussion about it can be found [here](https://aur.archlinux.org/packages/mpich/). The only thing I can do is `mpiexec.hydra` or `mpirun.hydra`, which invokes `mpich`.

Naturally, the consequence is that PETSc will fail due to incompatibilities between MPI implementation (`openmpi`) and Hydra process manager. See the log below.

[firedrake-install_change_mpiexec.log](https://github.com/firedrakeproject/firedrake/files/3849850/firedrake-install_change_mpiexec.log)

### Third attempt: isolated conda environment with Firedrake dependencies

My last resort. I don't like to mix `virtualenv` and `conda` for several reasons, but this way I can change/install things easily. I build a conda environment with the following packages (based on the install script recommendations when `--no-package-manager` is used):

```yaml
channels:
- conda-forge
dependencies:
- autoconf
- automake
- cmake
- compilers
- gcc_linux-64
- gfortran_linux-64
- gxx_linux-64
- libtool
- mpich
- python >=3.5
- zlib
name: firedrake-deps
```

Now, I ran the install script as usual, but with activated conda environment:

```console
$ python3 firedrake-install --disable-ssh --no-package-manager
```

But my bad luck is strong. No success. So I got intrigued and checked my compilers. 

TL;DR: here is the log -> [firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/3849905/firedrake-install.log)

Below, the results for my checks are provided:

For `mpiexec`:

```console
$ mpiexec --version
HYDRA build details:
    Version:                                 3.3.2
    Release Date:                            Tue Nov 12 21:23:16 CST 2019
    CC:                              x86_64-conda_cos6-linux-gnu-cc -I/home/volps/miniconda3/envs/firedrake-deps/include -I/home/volps/miniconda3/envs/firedrake-deps/include -L/home/volps/miniconda3/envs/firedrake-deps/lib -Wl,-rpath,/home/volps/miniconda3/envs/firedrake-deps/lib 
    CXX:                             x86_64-conda_cos6-linux-gnu-c++ -I/home/volps/miniconda3/envs/firedrake-deps/include -I/home/volps/miniconda3/envs/firedrake-deps/include -L/home/volps/miniconda3/envs/firedrake-deps/lib -Wl,-rpath,/home/volps/miniconda3/envs/firedrake-deps/lib 
    F77:                             x86_64-conda_cos6-linux-gnu-gfortran -I/home/volps/miniconda3/envs/firedrake-deps/include -L/home/volps/miniconda3/envs/firedrake-deps/lib -Wl,-rpath,/home/volps/miniconda3/envs/firedrake-deps/lib 
    F90:                             x86_64-conda_cos6-linux-gnu-gfortran -I/home/volps/miniconda3/envs/firedrake-deps/include -L/home/volps/miniconda3/envs/firedrake-deps/lib -Wl,-rpath,/home/volps/miniconda3/envs/firedrake-deps/lib 
    Configure options:                       '--disable-option-checking' '--prefix=/home/volps/miniconda3/envs/firedrake-deps' '--disable-dependency-tracking' '--enable-cxx' '--enable-fortran' '--disable-wrapper-rpath' 'MPICHLIB_CFLAGS=-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -I/home/volps/miniconda3/envs/firedrake-deps/include -fdebug-prefix-map=/home/conda/feedstock_root/build_artifacts/mpich-mpi_1573662435882/work=/usr/local/src/conda/mpich-3.3.2 -fdebug-prefix-map=/home/volps/miniconda3/envs/firedrake-deps=/usr/local/src/conda-prefix' 'MPICHLIB_CPPFLAGS=-DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -I/home/volps/miniconda3/envs/firedrake-deps/include' 'MPICHLIB_CXXFLAGS=-fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -I/home/volps/miniconda3/envs/firedrake-deps/include -fdebug-prefix-map=/home/conda/feedstock_root/build_artifacts/mpich-mpi_1573662435882/work=/usr/local/src/conda/mpich-3.3.2 -fdebug-prefix-map=/home/volps/miniconda3/envs/firedrake-deps=/usr/local/src/conda-prefix' 'MPICHLIB_FFLAGS=-fopenmp -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -I/home/volps/miniconda3/envs/firedrake-deps/include -fdebug-prefix-map=/home/conda/feedstock_root/build_artifacts/mpich-mpi_1573662435882/work=/usr/local/src/conda/mpich-3.3.2 -fdebug-prefix-map=/home/volps/miniconda3/envs/firedrake-deps=/usr/local/src/conda-prefix' 'MPICHLIB_FCFLAGS=-fopenmp -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -I/home/volps/miniconda3/envs/firedrake-deps/include -fdebug-prefix-map=/home/conda/feedstock_root/build_artifacts/mpich-mpi_1573662435882/work=/usr/local/src/conda/mpich-3.3.2 -fdebug-prefix-map=/home/volps/miniconda3/envs/firedrake-deps=/usr/local/src/conda-prefix' 'CC=x86_64-conda_cos6-linux-gnu-cc' 'CFLAGS=-I/home/volps/miniconda3/envs/firedrake-deps/include -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -I/home/volps/miniconda3/envs/firedrake-deps/include -fdebug-prefix-map=/home/conda/feedstock_root/build_artifacts/mpich-mpi_1573662435882/work=/usr/local/src/conda/mpich-3.3.2 -fdebug-prefix-map=/home/volps/miniconda3/envs/firedrake-deps=/usr/local/src/conda-prefix -O2' 'LDFLAGS=-L/home/volps/miniconda3/envs/firedrake-deps/lib -Wl,-rpath,/home/volps/miniconda3/envs/firedrake-deps/lib' 'CPPFLAGS=-I/home/volps/miniconda3/envs/firedrake-deps/include -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -I/home/volps/miniconda3/envs/firedrake-deps/include -I/home/conda/feedstock_root/build_artifacts/mpich-mpi_1573662435882/work/src/mpl/include -I/home/conda/feedstock_root/build_artifacts/mpich-mpi_1573662435882/work/src/mpl/include -I/home/conda/feedstock_root/build_artifacts/mpich-mpi_1573662435882/work/src/openpa/src -I/home/conda/feedstock_root/build_artifacts/mpich-mpi_1573662435882/work/src/openpa/src -D_REENTRANT -I/home/conda/feedstock_root/build_artifacts/mpich-mpi_1573662435882/work/src/mpi/romio/include' 'CPP=/home/volps/miniconda3/envs/firedrake-deps/bin/x86_64-conda_cos6-linux-gnu-cpp' 'CXX=x86_64-conda_cos6-linux-gnu-c++' 'CXXFLAGS=-I/home/volps/miniconda3/envs/firedrake-deps/include -fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -I/home/volps/miniconda3/envs/firedrake-deps/include -fdebug-prefix-map=/home/conda/feedstock_root/build_artifacts/mpich-mpi_1573662435882/work=/usr/local/src/conda/mpich-3.3.2 -fdebug-prefix-map=/home/volps/miniconda3/envs/firedrake-deps=/usr/local/src/conda-prefix -O2' 'FC=x86_64-conda_cos6-linux-gnu-gfortran' 'FCFLAGS=-I/home/volps/miniconda3/envs/firedrake-deps/include -fopenmp -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -I/home/volps/miniconda3/envs/firedrake-deps/include -fdebug-prefix-map=/home/conda/feedstock_root/build_artifacts/mpich-mpi_1573662435882/work=/usr/local/src/conda/mpich-3.3.2 -fdebug-prefix-map=/home/volps/miniconda3/envs/firedrake-deps=/usr/local/src/conda-prefix -O2' 'FFLAGS=-I/home/volps/miniconda3/envs/firedrake-deps/include -fopenmp -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -I/home/volps/miniconda3/envs/firedrake-deps/include -fdebug-prefix-map=/home/conda/feedstock_root/build_artifacts/mpich-mpi_1573662435882/work=/usr/local/src/conda/mpich-3.3.2 -fdebug-prefix-map=/home/volps/miniconda3/envs/firedrake-deps=/usr/local/src/conda-prefix -O2' '--cache-file=/dev/null' '--srcdir=.' 'LIBS=' 'MPLLIBNAME=mpl'
    Process Manager:                         pmi
    Launchers available:                     ssh rsh fork slurm ll lsf sge manual persist
    Topology libraries available:            hwloc
    Resource management kernels available:   user slurm ll lsf sge pbs cobalt
    Checkpointing libraries available:       
    Demux engines available:                 poll select
```

Another check, now for `mpicc`:

```console
$ mpicc -v         
mpicc for MPICH version 3.3.2
Reading specs from /home/volps/miniconda3/envs/firedrake-deps/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/specs
COLLECT_GCC=x86_64-conda_cos6-linux-gnu-cc
COLLECT_LTO_WRAPPER=/home/volps/miniconda3/envs/firedrake-deps/bin/../libexec/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/lto-wrapper
Target: x86_64-conda_cos6-linux-gnu
Configured with: /home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/src/gcc/configure --build=x86_64-build_pc-linux-gnu --host=x86_64-build_pc-linux-gnu --target=x86_64-conda_cos6-linux-gnu --prefix=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/gcc_built --with-sysroot=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/gcc_built/x86_64-conda_cos6-linux-gnu/sysroot --enable-languages=c,c++,fortran,objc,obj-c++ --with-pkgversion='crosstool-NG 1.23.0.452-d158' --enable-__cxa_atexit --disable-libmudflap --enable-libgomp --disable-libssp --enable-libquadmath --enable-libquadmath-support --enable-libsanitizer --enable-libmpx --with-gmp=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --with-mpfr=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --with-mpc=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --with-isl=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --enable-lto --with-host-libstdcxx='-static-libgcc -Wl,-Bstatic,-lstdc++,-Bdynamic -lm' --enable-threads=posix --enable-target-optspace --enable-plugin --enable-gold --disable-nls --disable-multilib --with-local-prefix=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/gcc_built/x86_64-conda_cos6-linux-gnu/sysroot --enable-long-long --enable-default-pie
Thread model: posix
gcc version 7.3.0 (crosstool-NG 1.23.0.452-d158)
```

And for `mpif90` as well:

```console
$ mpif90 -v
mpifort for MPICH version 3.3.2
Reading specs from /home/volps/miniconda3/envs/firedrake-deps/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/specs
Reading specs from /home/volps/miniconda3/envs/firedrake-deps/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/specs
COLLECT_GCC=/home/volps/miniconda3/envs/firedrake-deps/bin/x86_64-conda_cos6-linux-gnu-gfortran.bin
COLLECT_LTO_WRAPPER=/home/volps/miniconda3/envs/firedrake-deps/bin/../libexec/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/lto-wrapper
Target: x86_64-conda_cos6-linux-gnu
Configured with: /home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/src/gcc/configure --build=x86_64-build_pc-linux-gnu --host=x86_64-build_pc-linux-gnu --target=x86_64-conda_cos6-linux-gnu --prefix=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/gcc_built --with-sysroot=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/gcc_built/x86_64-conda_cos6-linux-gnu/sysroot --enable-languages=c,c++,fortran,objc,obj-c++ --with-pkgversion='crosstool-NG 1.23.0.452-d158' --enable-__cxa_atexit --disable-libmudflap --enable-libgomp --disable-libssp --enable-libquadmath --enable-libquadmath-support --enable-libsanitizer --enable-libmpx --with-gmp=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --with-mpfr=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --with-mpc=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --with-isl=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --enable-lto --with-host-libstdcxx='-static-libgcc -Wl,-Bstatic,-lstdc++,-Bdynamic -lm' --enable-threads=posix --enable-target-optspace --enable-plugin --enable-gold --disable-nls --disable-multilib --with-local-prefix=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/gcc_built/x86_64-conda_cos6-linux-gnu/sysroot --enable-long-long --enable-default-pie
Thread model: posix
gcc version 7.3.0 (crosstool-NG 1.23.0.452-d158) 
```

Now, for the compilers:

* gcc

```console
$ gcc -v     
Reading specs from /home/volps/miniconda3/envs/firedrake-deps/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/specs
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/home/volps/miniconda3/envs/firedrake-deps/bin/../libexec/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/lto-wrapper
Target: x86_64-conda_cos6-linux-gnu
Configured with: /home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/src/gcc/configure --build=x86_64-build_pc-linux-gnu --host=x86_64-build_pc-linux-gnu --target=x86_64-conda_cos6-linux-gnu --prefix=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/gcc_built --with-sysroot=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/gcc_built/x86_64-conda_cos6-linux-gnu/sysroot --enable-languages=c,c++,fortran,objc,obj-c++ --with-pkgversion='crosstool-NG 1.23.0.452-d158' --enable-__cxa_atexit --disable-libmudflap --enable-libgomp --disable-libssp --enable-libquadmath --enable-libquadmath-support --enable-libsanitizer --enable-libmpx --with-gmp=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --with-mpfr=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --with-mpc=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --with-isl=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --enable-lto --with-host-libstdcxx='-static-libgcc -Wl,-Bstatic,-lstdc++,-Bdynamic -lm' --enable-threads=posix --enable-target-optspace --enable-plugin --enable-gold --disable-nls --disable-multilib --with-local-prefix=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/gcc_built/x86_64-conda_cos6-linux-gnu/sysroot --enable-long-long --enable-default-pie
Thread model: posix
gcc version 7.3.0 (crosstool-NG 1.23.0.452-d158)
```

* `gfortran`

```console
$ gfortran -v
Reading specs from /home/volps/miniconda3/envs/firedrake-deps/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/specs
Reading specs from /home/volps/miniconda3/envs/firedrake-deps/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/specs
COLLECT_GCC=/home/volps/miniconda3/envs/firedrake-deps/bin/x86_64-conda_cos6-linux-gnu-gfortran.bin
COLLECT_LTO_WRAPPER=/home/volps/miniconda3/envs/firedrake-deps/bin/../libexec/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/lto-wrapper
Target: x86_64-conda_cos6-linux-gnu
Configured with: /home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/src/gcc/configure --build=x86_64-build_pc-linux-gnu --host=x86_64-build_pc-linux-gnu --target=x86_64-conda_cos6-linux-gnu --prefix=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/gcc_built --with-sysroot=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/gcc_built/x86_64-conda_cos6-linux-gnu/sysroot --enable-languages=c,c++,fortran,objc,obj-c++ --with-pkgversion='crosstool-NG 1.23.0.452-d158' --enable-__cxa_atexit --disable-libmudflap --enable-libgomp --disable-libssp --enable-libquadmath --enable-libquadmath-support --enable-libsanitizer --enable-libmpx --with-gmp=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --with-mpfr=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --with-mpc=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --with-isl=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --enable-lto --with-host-libstdcxx='-static-libgcc -Wl,-Bstatic,-lstdc++,-Bdynamic -lm' --enable-threads=posix --enable-target-optspace --enable-plugin --enable-gold --disable-nls --disable-multilib --with-local-prefix=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/gcc_built/x86_64-conda_cos6-linux-gnu/sysroot --enable-long-long --enable-default-pie
Thread model: posix
gcc version 7.3.0 (crosstool-NG 1.23.0.452-d158)
```

For me, everything looks fine. I'm unable to detect what is wrong.

Please, could you give me some help?

Thanks a lot!
Diego.

**Comment by wence-** [comment]
Option 1:

This is unfortunately a known breakage: pnetcdf uses certain features of MPI-1 (that were deprecated when MPI-2 was released in 1999). OpenMPI-4 now finally switches those features off, so you get build errors.

For option 2:

> python3 firedrake-install --disable-ssh --no-package-manager --mpiexec mpiexec.hydra --mpicc mpicc --mpicxx mpicxx --mpif90 mpif90

You can also say `--mpicc mpicc.mpich ...` so that you get consistency between mpiexec and mpicc etc... We then put appropriate symlinks into the virtualenv path, so when you activate it, you get the right binaries in your environment.

For option 3:

No idea. Can you compile and then run a trivial C file with the conda environment?

```
#include <mpi.h>
#include <stdio.h>
int main(void)
{
   int rank, size;
   MPI_Init(NULL, NULL);
   MPI_Comm_rank(MPI_COMM_WORLD, &rank);
   MPI_Comm_Size(MPI_COMM_WORLD, &size);
   printf("Rank %d; Size %d\n", rank, size);
   MPI_Finalize();
   return 0;
}
```

**Comment by wence-** [comment]
Aside, as part of these problems, @dham is working on getting firedrake-install (via PETSc) to build MPICH for us.

**Comment by volpatto** [comment]
Thanks for reply, @wence-.

Great to know that David is on a MPICH for Firedrake. This that users (probably) will have it soon.

I thought a little about what you said in the below quote, and I finally solved my installation issue.

> You can also say `--mpicc mpicc.mpich ...` so that you get consistency between mpiexec and mpicc etc... We then put appropriate symlinks into the virtualenv path, so when you activate it, you get the right binaries in your environment.

But, please, let me write down some remarks in order to help Firedrake users that eventually face the same problem. Also, I think that it's good to have this issue recorded for future references.

About `--mpicc mpicc.mpich ...` (for readers, a reminder: you have to provide `mpiexec`, `mpicc`, `mpicxx` and `mpif90` at once), this works for Ubuntu (maybe Debian-like OSs?), at least when I was a Ubuntu user. For Archlinux-based OS, the things are slightly different. There, you don't have `mpicc.mpich` provided for you. You only can access `mpirun`/`mpiexec` with Hydra. Moreover, MPICH is installed in `/opt/mpich` and added to your path with a lower priority when compared with `openmpi`. It seems that this is necessary because Archlinux depends on `openmpi`, so it can be dangerous if you change priorities for different MPI implementations.

So, in order to successfully install Firedrake with consistent MPICH, you can create yourself symlinks `mpicc.mpich` and etc. You can find everything inside `/opt/mpich/bin`. Or you can pass it directly as `firedrake-install` script options. So, if you don't want to create symlinks for any reason, or if you just want to do a "copy-paste" solution, here is the command that successfully installed Firedrake for me (Manjaro 18.1.3 OS) circumventing PNETCDF issue:

```console
$ python3 firedrake-install --disable-ssh --no-package-manager --mpiexec /opt/mpich/bin/mpiexec --mpicc /opt/mpich/bin/mpicc --mpicxx /opt/mpich/bin/mpicxx --mpif90 /opt/mpich/bin/mpifort
```

But what about other GNU/Linux? The key ingredient to solve PNETCDF seems to be a proper usage of MPICH, as already pointed by @wence-, due to deprecated features that are removed in a recent OpenMPI release. So, if you are using other OS which is not Archlinux-based or Ubuntu/Debian-based, you just have to install compatible MPICH (in my case, it was MPICH version 3.3.1), and pass the path to the bins for `mpiexec`, `mpicc`, `mpicxx` and `mpif90` to the `firedrake-install` script.

That's all. I hope that it can help.

**Comment by volpatto** [comment]
@wence- thanks for the help again! 

I'm closing this issue since the it's solved.

**Comment by wence-** [comment]
@volpatto Would you mind preparing a patch for the documentation to add this as an FAQ in the installation page?

**Comment by volpatto** [comment]
> @volpatto Would you mind preparing a patch for the documentation to add this as an FAQ in the installation page?

Sure, I can do it. I think that I can open a PR until next Monday, if this time range is acceptable. What do you think?

**Comment by wence-** [comment]
Yes, no problem! (Have you seen how long it takes me to do any code review at the moment? You're not likely to be the rate-limiting step).

**Comment by volpatto** [comment]
> Yes, no problem! (Have you seen how long it takes me to do any code review at the moment? You're not likely to be the rate-limiting step).

Yeah, I can imagine that! A lot of PRs here, and I assume that you have other projects (FInAT, some PETSc things and others).

Okay, then. I will begin to adapt things from here to the documentation ASAP.
